receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go

```diff
@@ -0,0 +1,248 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package host
+
+import (
+	"bufio"
+	"fmt"
+	"hash/fnv"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"go.uber.org/zap"
+)
+
+type EbsVolume struct {
+	// device name to volumeID mapping
```

Comment:
You can put the mutex above the fields it is protecting, e.g. 

```go
// no need to protect because they are readonly or has internal protection
refershInterval time.Duration
logger *zap.Logger

mu sync.RWMutex
dev2Vol map[string]string
```

And I don't think you need to embed sync.RWMutex

Reply:
Right. That's better.